src/conversations/Conversations.ts

```diff
@@ -36,34 +43,45 @@ export default class Conversations {
       const peerAddress = this.getPeerAddress(message)
 
       if (peerAddress) {
-        seenPeers.add(peerAddress)
+        const have = seenPeers.get(peerAddress)
+        if (!have || have < message.sent) {
+          seenPeers.set(peerAddress, message.sent)
+        }
       }
     }
 
-    return (
-      Array.from(seenPeers)
-        // Consistently order the results
-        .sort()
-        .map((peerAddress) => new Conversation(this.client, peerAddress))
+    const conversations: Conversation[] = []
+    seenPeers.forEach((sent, peerAddress) =>
+      conversations.push(new ConversationV1(this.client, peerAddress, sent))
     )
+
+    for (const sealed of await this.client.listInvitations()) {
+      const unsealed = await sealed.v1.getInvitation(this.client.keys)
+      conversations.push(
+        new ConversationV2(this.client, unsealed, sealed.v1.header)
+      )
+    }
+
+    conversations.sort((a, b) => a.sent.getTime() - b.sent.getTime())
+    return conversations
   }
 
   /**
    * Returns a stream of any newly created conversations.
    * Will dedupe to not return the same conversation twice in the same stream.
    * Does not dedupe any other previously seen conversations
    */
-  stream(): Promise<Stream<Conversation>> {
-    const messageTransformer: MessageTransformer<Conversation> = (
-      msg: Message
+  async stream(): Promise<JoinStream<ConversationV1, ConversationV2>> {
```

Comment:
It seems like this would be more efficient to just stream from both topics, and have a switch in the decoding step that would either create a `ConversationV1` or a `ConversationV2` depending on the topic, rather than keeping 2 HTTP connections open

Reply:
That sounds a lot better. This is quite a mess at the moment 👍

Reply:
This turned out to be soooo much simpler! ❤️ 🙇 🙏